🐛 Bug: release-it failures in repos: undefined

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Once a newly created package has a valid `NPM_TOKEN` secret, it should be able to publish.

### Actual

I've noticed this start to fail in new packages. Example from https://github.com/JoshuaKGoldberg/octokit-from-auth/actions/runs/12340223056/job/34437399138:

```plaintext
Start: running release-it
Error: Error running release-it: Error: Command failed with ENOENT: release-it --verbose
spawn release-it ENOENT
Error: Error: Command failed with ENOENT: release-it --verbose
spawn release-it ENOENT
Result from running release-it: undefined
```

...huh?

### Additional Info

💖

```plaintext
$ npx release-it --verbose
ERROR Cannot find module '@release-it/conventional-changelog'
Require stack:
- /Users/josh/repos/octokit-from-auth
```

--- a/src/next/blocks/blockReleaseIt.ts
+++ b/src/next/blocks/blockReleaseIt.ts
@@ -2,6 +2,7 @@ import { createSoloWorkflowFile } from "../../steps/writing/creation/dotGitHub/c
 import { base } from "../base.js";
 import { blockCSpell } from "./blockCSpell.js";
 import { blockPackageJson } from "./blockPackageJson.js";
+import { getPackageDependencies } from "./packageData.js";
 
 export const blockReleaseIt = base.createBlock({
 	about: {
@@ -15,6 +16,10 @@ export const blockReleaseIt = base.createBlock({
 				}),
 				blockPackageJson({
 					properties: {
+						devDependencies: getPackageDependencies(
+							"@release-it/conventional-changelog",
+							"release-it",
+						),
 						publishConfig: {
 							provenance: true,
 						},

